c++/src/kj/thread.c++

```diff
@@ -33,8 +33,8 @@ namespace kj {
 
 #if _WIN32
 
-Thread::Thread(Function<void()> func): func(kj::mv(func)) {
-  threadHandle = CreateThread(nullptr, 0, &runThread, this, 0, nullptr);
+Thread::Thread(Function<void()> func): state(new ThreadState { kj::mv(func), nullptr, 2 }) {
+  threadHandle = CreateThread(nullptr, 0, &runThread, state, 0, nullptr);
   KJ_ASSERT(threadHandle != nullptr, "CreateThread failed.");
```

Comment:
Probably need a

    if( threadHandle==nullptr ) state->unref();

here to do the other thread's cleanup if it doesn't start.  I don't know if CreateThread can throw - if it does this gets more complicated.